analyzer/codechecker_analyzer/analyzers/config_handler.py

```diff
@@ -214,31 +214,23 @@ def initialize_checkers(self,
         reserved_names = self.__gen_name_variations()
 
         for identifier, enabled in cmdline_enable:
-            if identifier.startswith('profile:'):
-                profile_name = identifier[len('profile:'):]
-                if profile_name not in profile_map.available_profiles():
-                    LOG.error('No such profile: %s', profile_name)
-                    sys.exit(1)
-                for checker in profile_map.by_profile(profile_name):
+            if ':' in identifier:
+                for checker in checker_labels.checkers_by_labels([identifier]):
                     self.set_checker_enabled(checker, enabled)
-            if identifier.startswith('guideline:'):
-                guideline_name = identifier[len('guideline:'):]
-                if guideline_name not in guideline_map.available_guidelines():
-                    LOG.error('No such guideline: %s', guideline_name)
-                    sys.exit(1)
-                for checker in guideline_map.by_guideline(guideline_name):
-                    self.set_checker_enabled(checker, enabled)
-            elif identifier in profile_map.available_profiles():
+            elif identifier in checker_labels.get_constraint(
+                    'profile', 'choice'):
```

Comment:
We are in a for loop, so this `get_constraint` function call will be called multiple times but it will return the same value every time. So we should call it only once.

Reply:
This is proportional to the number of `--enable`/`--disable` flags and this is a constant operation. I don't think if there is any measurable difference, but if you insist, I can factor it out.